stake-o-matic/src/confirmed_block_cache.rs

```diff
@@ -0,0 +1,203 @@
+use crate::retry_rpc_operation;
+use serde::{Deserialize, Serialize};
+use solana_client::rpc_client::RpcClient;
+use solana_sdk::{clock::Slot, commitment_config::CommitmentConfig, epoch_info::EpochInfo};
+use std::{
+    cell::RefCell,
+    fs::{self, File, OpenOptions},
+    io,
+    ops::Range,
+    path::{Path, PathBuf},
+};
+
+#[derive(Clone, Debug, Default)]
+struct Entry {
+    slots: Range<Slot>,
+    path: PathBuf,
+}
+
+impl Entry {
+    pub fn new<P: AsRef<Path>>(base_path: P, slots: Range<Slot>) -> Self {
+        let file_name = format!("{}-{}.json", slots.start, slots.end);
+        let path = base_path.as_ref().join(file_name);
+        Self { slots, path }
+    }
+
+    fn parse_filename<F: AsRef<str>>(filename: F) -> Option<Range<Slot>> {
+        let filename = filename.as_ref();
+        let mut parts = filename.splitn(2, '.');
```

Comment:
Use [extension()](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) to avoid the manual parsing?